### DISCUSSION TITLE
Require "components.d.ts" for built-in component completion? (e.g. <NuxtLink />)

### DISCUSSION THREAD

**Discussion by yaegassy** [nuxt/nuxt#17959] [Questions] [answered]
I have checked this PR and issue.

- https://github.com/nuxt/framework/pull/2340
- https://github.com/nuxt/framework/issues/2329

In this PR, it seem like `<NuxtLink />` is already available without `components.d.ts`.

Is there any mistake in my interpretation? Or do I need to add some settings to `nuxt.config.ts`?

In my environment, I can't completion <NuxtLink /> etc. without `components.d.ts`.

### DEMO (mp4)

> Install nuxt3 with `nuxi` and `npm run dev` has already been executed.

https://user-images.githubusercontent.com/188642/152913875-fa336cbe-ee9d-4019-9e9e-3455a68365ef.mp4

**Comment by danielroe** [comment, marked as answer]
You will only get type completion for those components if you have a `pages/` directory in your app, as they are only provided globally in that case.

**Reply by yaegassy** [reply]
Thanks for the reply! My question has been solved. 🙇
